include/arch/x86/intel64/arch.h

```diff
@@ -93,11 +93,20 @@ struct x86_ssf {
 
 #endif /* _ASMLANGUAGE */
 
+#ifdef CONFIG_PCIE
+#define X86_RESERVE_IRQ(irq_p, name) \
+	static Z_DECL_ALIGN(uint8_t) name \
+	__in_section(_irq_alloc, static, name) __used = irq_p
+#else
+#define X86_RESERVE_IRQ(irq_p, name)
+#endif
+
 /*
  * All Intel64 interrupts are dynamically connected.
  */
 
 #define ARCH_IRQ_CONNECT(irq_p, priority_p, isr_p, isr_param_p, flags_p) \
+	X86_RESERVE_IRQ(irq_p, _CONCAT(_irq_alloc_fixed, __COUNTER__)); \
```

Comment:
IIRC, `__COUNTER__` starts as 0 for each source file. Multiple drivers may result in multiple `_irq_alloc_fixed0` (and so on).

Reply:
That's a fair point, but it appears it's not an issue since we never use these variables by name but by array index. I managed to link an image where both HPET and GPIO had the same `_irq_alloc_fixed0` name without any compiler or linker issues:
```
irq_alloc       0x000000000010b130        0x2                                   
                0x000000000010b130                __irq_alloc_start = .         
 *(SORT_BY_NAME(SORT_BY_ALIGNMENT(._irq_alloc*)))                               
 ._irq_alloc.static._irq_alloc_fixed0                                           
                0x000000000010b130        0x1 zephyr/libzephyr.a(hpet.c.obj)    
 ._irq_alloc.static._irq_alloc_fixed0                                           
                0x000000000010b131        0x1 zephyr/drivers/gpio/libdrivers__gpio.a(gpio_intel_apl.c.obj)
                0x000000000010b132                __irq_alloc_end = .   
```

So I'd conclude that it's only important that these are unique within each compilation unit (which `__COUNTER__` does guarantee).